src/output.rs

```diff
@@ -369,75 +370,94 @@ fn format_bytes(data: &[u8], f: &mut impl fmt::Write) -> fmt::Result {
     assert!(BYTES_MAX_PRINTED < BYTES_MIN_OVERFLOW);
 
     let lines_total = data.as_bstr().lines().count();
+    let multiline = 1 < lines_total;
 
-    if lines_total >= LINES_MIN_OVERFLOW {
+    if LINES_MIN_OVERFLOW <= lines_total {
         let lines_omitted = lines_total - LINES_MAX_PRINTED;
         let start_lines = data.as_bstr().lines().take(LINES_MAX_START);
         let end_lines = data.as_bstr().lines().skip(LINES_MAX_START + lines_omitted);
-        write!(f, "<{} lines total>", lines_total)?;
-        write_debug_bstrs(f, start_lines)?;
-        write!(f, "\n<{} lines omitted>\n", lines_omitted)?;
-        write_debug_bstrs(f, end_lines)
-    } else if data.len() >= BYTES_MIN_OVERFLOW {
-        write!(f, "<{} bytes total>", data.len())?;
-        write_debug_bstrs(f, data[..BYTES_MAX_START].lines())?;
-        write!(f, "<{} bytes omitted>", data.len() - BYTES_MAX_PRINTED)?;
-        write_debug_bstrs(f, data[data.len() - BYTES_MAX_END..].lines())
+        writeln!(f, "<{} lines total>", lines_total)?;
+        write_debug_bstrs(f, true, start_lines)?;
+        writeln!(f, "<{} lines omitted>", lines_omitted)?;
+        write_debug_bstrs(f, true, end_lines)
+    } else if BYTES_MIN_OVERFLOW <= data.len() {
+        write!(
+            f,
+            "<{} bytes total>{}",
+            data.len(),
+            if multiline { "\n" } else { "" }
+        )?;
+        write_debug_bstrs(f, multiline, data[..BYTES_MAX_START].lines())?;
+        write!(
+            f,
+            "<{} bytes omitted>{}",
+            data.len() - BYTES_MAX_PRINTED,
+            if multiline { "\n" } else { "" }
+        )?;
+        write_debug_bstrs(f, multiline, data[data.len() - BYTES_MAX_END..].lines())
     } else {
-        write_debug_bstrs(f, data.lines())
+        write_debug_bstrs(f, multiline, data.lines())
```

Comment:
Probably not a blocker but one wrinkle in all of this I just remembered is that trailing newlines are a subtle difference in output that is hard to spot and we are normalizing it with this output mode.

One option is to use bstr's `lines_with_terminator()` and rely on that for the newlines but still add a final newline. 

snapbox shows a `∅` when there isn't a trailing newline but that could get complicated here with how we reuse `write_debug_bstrs`

Reply:
> One option is to use bstr's `lines_with_terminator()` and rely on that for the newlines but still add a final newline.

Sorry, I am a little confused. Are you you suggesting that we display the `\n` so that it is apparent when the last one is missing?

@tailhook gave a [reason](https://github.com/assert-rs/assert_cmd/issues/137#issuecomment-1354732014) to not include both `\n` and linebreaks.

Reply:
The idea is you if you use `lines_with_terminator` then you can remove
```rust
            if multiline {
                writeln!(f)?;
            }
``
You then split off the line terminator and write out a real line terminator rather than the dbg output of one

Reply:
Please tell me if this is what you had in mind.

Reply:
Basically, thanks

Reply:
Thank you!